shared/reducers/teams.tsx

```diff
@@ -82,7 +82,19 @@ export default Container.makeReducer<
     draftState.teamNameToLoadingInvites.set(teamname, oldLoadingInvites)
   },
   [TeamsGen.teamLoaded]: (draftState, action) => {
-    const {teamID, details} = action.payload
+    const {teamID, team} = action.payload
+    const maybeMeta = draftState.teamMeta.get(teamID)
+    if (maybeMeta && maybeMeta.teamname !== team.name) {
+      if (team.name.includes('.')) {
+        // subteam name changed. store loaded name
+        maybeMeta.teamname = team.name
```

Comment:
I understand why this is necessary but it's unfortunate

Better would be to pass in the expected team name to the rpc, even better would be to not send the teamid in the first place. Either way I think this PR is good for now, thanks for taking a look

Reply:
Would getting verified team names in the list be harder than adding guards? I still think indexing by teamid makes sense.